drivers/video/arducam_mega.c

```diff
@@ -0,0 +1,1048 @@
+}
+
+static int arducam_mega_set_exposure(const struct device *dev, uint32_t value)
+{
+	int ret = 0;
+	const struct arducam_mega_config *cfg = dev->config;
+
+	ret |= arducam_mega_await_bus_idle(&cfg->bus, 3);
+	ret |= arducam_mega_write_reg(&cfg->bus, CAM_REG_MANUAL_EXPOSURE_BIT_19_16,
+				      (value >> 16) & 0xff);
+	ret |= arducam_mega_await_bus_idle(&cfg->bus, 10);
+	ret |= arducam_mega_write_reg(&cfg->bus, CAM_REG_MANUAL_EXPOSURE_BIT_15_8,
+				      (value >> 8) & 0xff);
+	ret |= arducam_mega_await_bus_idle(&cfg->bus, 10);
+	ret |= arducam_mega_write_reg(&cfg->bus, CAM_REG_MANUAL_EXPOSURE_BIT_7_0, value & 0xff);
+	ret |= arducam_mega_await_bus_idle(&cfg->bus, 10);
+	return ret;
+}
+
+static int arducam_mega_set_resolution(const struct device *dev, enum MEGA_RESOLUTION resolution)
+{
+	int ret = 0;
+	const struct arducam_mega_config *cfg = dev->config;
+
+	ret |= arducam_mega_await_bus_idle(&cfg->bus, 3);
+
+	ret |= arducam_mega_write_reg(&cfg->bus, CAM_REG_CAPTURE_RESOLUTION, resolution);
+
+	ret |= arducam_mega_await_bus_idle(&cfg->bus, 10);
+
+	return ret;
+}
+
+static int arducam_mega_check_connection(const struct device *dev)
+{
+	int ret = 0;
+	uint8_t cam_id;
+	const struct arducam_mega_config *cfg = dev->config;
+	struct arducam_mega_data *drv_data = dev->data;
+
+	ret |= arducam_mega_await_bus_idle(&cfg->bus, 255);
+	cam_id = arducam_mega_read_reg(&cfg->bus, CAM_REG_SENSOR_ID);
+
+	if (!(cam_id & 0x87)) {
+		LOG_ERR("arducam mega not detected, 0x%x\n", cam_id);
+		return -ENODEV;
+	}
+
+	LOG_INF("detect camera id 0x%x, ret = %d\n", cam_id, ret);
+
+	switch (cam_id) {
+	case ARDUCAM_SENSOR_5MP_1: /* 5MP-1 */
+		fmts[8] = (struct video_format_cap)ARDUCAM_MEGA_VIDEO_FORMAT_CAP(
+			2592, 1944, VIDEO_PIX_FMT_RGB565);
+		fmts[17] = (struct video_format_cap)ARDUCAM_MEGA_VIDEO_FORMAT_CAP(
+			2592, 1944, VIDEO_PIX_FMT_JPEG);
+		fmts[26] = (struct video_format_cap)ARDUCAM_MEGA_VIDEO_FORMAT_CAP(
+			2592, 1944, VIDEO_PIX_FMT_YUYV);
+		support_resolution[8] = MEGA_RESOLUTION_WQXGA2;
+		drv_data->info = &mega_infos[0];
+		break;
+	case ARDUCAM_SENSOR_3MP_1: /* 3MP-1 */
+		fmts[8] = (struct video_format_cap)ARDUCAM_MEGA_VIDEO_FORMAT_CAP(
+			2048, 1536, VIDEO_PIX_FMT_RGB565);
+		fmts[17] = (struct video_format_cap)ARDUCAM_MEGA_VIDEO_FORMAT_CAP(
+			2048, 1536, VIDEO_PIX_FMT_JPEG);
+		fmts[26] = (struct video_format_cap)ARDUCAM_MEGA_VIDEO_FORMAT_CAP(
+			2048, 1536, VIDEO_PIX_FMT_YUYV);
+		support_resolution[8] = MEGA_RESOLUTION_QXGA;
+		drv_data->info = &mega_infos[1];
+		break;
+	case ARDUCAM_SENSOR_5MP_2: /* 5MP-2 */
+		fmts[8] = (struct video_format_cap)ARDUCAM_MEGA_VIDEO_FORMAT_CAP(
+			2592, 1936, VIDEO_PIX_FMT_RGB565);
+		fmts[17] = (struct video_format_cap)ARDUCAM_MEGA_VIDEO_FORMAT_CAP(
+			2592, 1936, VIDEO_PIX_FMT_JPEG);
+		fmts[26] = (struct video_format_cap)ARDUCAM_MEGA_VIDEO_FORMAT_CAP(
+			2592, 1936, VIDEO_PIX_FMT_YUYV);
+		support_resolution[8] = MEGA_RESOLUTION_WQXGA2;
+		break;
+		drv_data->info = &mega_infos[0];
```

Comment:
I think this code would never be executed.
This looks like a bug in the original driver maybe.

```suggestion
		drv_data->info = &mega_infos[0];
```